packages/library-base/src/nativeDarwin/kotlin/io/realm/kotlin/internal/platform/SystemUtils.kt

```diff
@@ -57,16 +57,16 @@ public actual fun epochInSeconds(): Long =
 @Suppress("MagicNumber")
 internal actual fun currentTime(): RealmInstant {
     val secs = NSDate().timeIntervalSince1970
-    val millis = (secs * 1000 + if (secs > 0) 0.5 else -0.5).toLong()
-    return if (millis < RealmInstant.MIN.epochSeconds * 1000) {
-        RealmInstant.MIN
-    } else if (millis > RealmInstant.MAX.epochSeconds * 1000) {
-        RealmInstant.MAX
-    } else {
-        RealmInstantImpl(
-            millis.floorDiv(1000.toLong()),
-            (millis.mod(1000.toLong()) * 1000).toInt()
-        )
+    return when {
+        // We can't convert the MIN value to ms as it will cause Long overflow so we have to compare directly against seconds
```

Comment:
IMO, I would be a bit uncomfortable merging this without unit tests, which means we need to refactor this method so we can inject test values.

Looking at the issue, I'm not sure I see how this was causing iOS to overflow. That doesn't feel like that should be possible for reasonable values of NSDate?

Reply:
That's fair. I will add a test that checks the value is not repeated, which was the case before. See explanation above for overflow.